client/state-db/src/lib.rs

```diff
@@ -115,9 +142,19 @@ pub trait NodeDb {
 }
 
 /// Error type.
-pub enum Error<E: fmt::Debug> {
+pub enum Error<E> {
 	/// Database backend error.
 	Db(E),
+
+	StateDbError(StateDbError),
+}
+impl<E> From<StateDbError> for Error<E> {
```

Comment:
```suggestion

impl<E> From<StateDbError> for Error<E> {
```

nitpick, feel free to ignore